### DISCUSSION TITLE
Redash Initial Setup stuck

### DISCUSSION THREAD

**Discussion by gaecoli** [getredash/redash#7148] [Q&A] [answered]
When i use ubuntu20 with docker to start redash, it show to me this.
<img width="827" alt="image" src="https://github.com/user-attachments/assets/a7d42074-7f94-4e84-a5e3-28885c42d59b">

**Comment by justinclift** [comment]
Oh.  That looks like a docker image that doesn't have the client side (ie front end) pieces included.

How was the image created?

**Reply by gaecoli** [reply]
> Oh. That looks like a docker image that doesn't have the client side (ie front end) pieces included.
> 
> How was the image created?

I run `make compose_build`, maybe i lost some command?

**Reply by justinclift** [reply, marked as answer]
Yeah.  I *think* you forgot to run `make build` before that.  The `make build` command builds the front end, then the `make compose_build` builds the image and pulls in the pieces that `make build` generated.

I wonder if it makes sense for us to have just one command that does both?

**Reply by gaecoli** [reply]
> Yeah. I _think_ you forgot to run `make build` before that. The `make build` command builds the front end, then the `make compose_build` builds the image and pulls in the pieces that `make build` generated.
> 
> I wonder if it makes sense for us to have just one command that does both?

Thank u!!!! But why we build without include front end? Before V10 i rememeber we run `docker build` it can include front end.

**Reply by justinclift** [reply]
I really don't remember.  From rough memory there's a use to having a docker image with only the Python backend in it, though I don't remember what that use is as I don't really do front end development of Redash (yet).

**Reply by justinclift** [reply]
As a general idea, I tend to rely on having an appropriate "local development setup" wiki page open when I'm working on Redash stuff:

https://github.com/getredash/redash/wiki

That helps me ensure I don't miss steps when doing things, as there can be a lot of distractions happening through a day.

Not sure if that approach is useful for you, but it might be worth considering. :smile:

**Reply by gaecoli** [reply]
> As a general idea, I tend to rely on having an appropriate "local development setup" wiki page open when I'm working on Redash stuff:
> 
> https://github.com/getredash/redash/wiki
> 
> That helps me ensure I don't miss steps when doing things, as there can be a lot of distractions happening through a day.
> 
> Not sure if that approach is useful for you, but it might be worth considering. 😄

Thank you!
